File: fieldsync-inventory-backend/src/main/java/com/example/fieldsync_inventory_api/entity/procurement/SupplierTypeEntity.java
package com.example.fieldsync_inventory_api.entity.procurement;

import jakarta.persistence.*;
import lombok.*;
import com.example.fieldsync_inventory_api.entity.Auditable;

import java.time.Instant;

@Entity
@Table(name = "supplier_type")
@Getter
@Setter
@ToString(callSuper = true)
@EqualsAndHashCode(callSuper = true)
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class SupplierTypeEntity extends Auditable {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Integer id;

    @Column(length = 50, unique = true, nullable = false)
    private String name;

    @Column(columnDefinition = "TEXT", nullable = false)
    private String description;

    @Column(name = "deleted_at")
    private Instant deletedAt;
}

File: fieldsync-inventory-app/app/src/main/java/com/example/fieldsync_inventory_app/data/local/entity/SupplierTypeEntity.kt
package com.example.fieldsync_inventory_app.data.local.entity

import androidx.room.Entity
import androidx.room.PrimaryKey

@Entity(tableName = "supplier_type")
data class SupplierTypeEntity(
    @PrimaryKey
    val id: Int,
    val name: String,
    val description: String,
)